src/runtime/auto-prompt-manager.js

```diff
@@ -365,56 +376,62 @@ export class AutoPromptManager {
    *   dismissedPrompts: (?string|undefined),
    *   shouldShowAutoPrompt: (boolean|undefined),
    * }} params
-   * @return {!string|undefined}
+   * @return {!Promise<string|undefined>}
    */
   getAudienceActionPromptType_({
     article,
     autoPromptType,
     dismissedPrompts,
     shouldShowAutoPrompt,
   }) {
-    let potentialActions = article?.audienceActions?.actions || [];
-    potentialActions = potentialActions.filter((action) =>
-      this.checkActionEligibility_(action.type)
-    );
-
-    // No audience actions means use the default prompt.
-    if (potentialActions.length === 0) {
-      return undefined;
-    }
-
-    // Default to the first recommended action.
-    let actionToUse = potentialActions[0].type;
-
-    // Contribution prompts should appear before recommended actions, so we'll need
-    // to check if we have shown it before.
-    if (
-      autoPromptType === AutoPromptType.CONTRIBUTION ||
-      autoPromptType === AutoPromptType.CONTRIBUTION_LARGE
-    ) {
-      if (!dismissedPrompts) {
-        this.promptDisplayed_ = AutoPromptType.CONTRIBUTION;
-        return undefined;
-      }
-      const previousPrompts = dismissedPrompts.split(',');
-      potentialActions = potentialActions.filter(
-        (action) => !previousPrompts.includes(action.type)
+    const audienceActions = article?.audienceActions?.actions || [];
+
+    return this.getEvent_(
+      COMPLETED_ACTION_TO_STORAGE_KEY_MAP.get(
+        AnalyticsEvent.ACTION_SURVEY_DATA_TRANSFER
+      )
+    ).then((surveysCompleted) => {
```

Comment:
```suggestion
    ).then((surveysCompletionTimestamps) => {
      const surveysCompleted = surveysCompletionTimestamps.length;
```

Reply:
Also renamed `completedSurvey` to `hasCompletedSurveys` for clarity.

Reply:
Awesome